firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+TabToolbarDelegate.swift

```diff
@@ -198,48 +198,80 @@ extension BrowserViewController: TabToolbarDelegate, PhotonActionSheetProtocol {
+
         return [privateBrowsingMode]
     }
 
     func getMoreTabToolbarLongPressActions() -> [PhotonRowActions] {
-        let newTab = SingleActionViewModel(title: .KeyboardShortcuts.NewTab,
-                                           iconString: StandardImageIdentifiers.Large.plus,
-                                           iconType: .Image) { _ in
+        let newTab = getNewTabAction()
+        let newPrivateTab = getNewPrivateTabAction()
+        let closeTab = getCloseTabAction()
+
+        if let tab = self.tabManager.selectedTab {
+            return tab.isPrivate ? [newPrivateTab, closeTab] : [newTab, closeTab]
+        }
+
+        return [newTab, closeTab]
+    }
+
+    func getTabToolbarRefactorLongPressActions() -> [[PhotonRowActions]] {
+        let newTab = getNewTabAction()
+        let newPrivateTab = getNewPrivateTabAction()
+        let closeTab = getCloseTabAction()
+
+        return [[newTab, newPrivateTab], [closeTab]]
+    }
+
+    func getNewTabLongPressActions() -> [[PhotonRowActions]] {
+        let newTab = getNewTabAction()
+        let newPrivateTab = getNewPrivateTabAction()
+
+        return [[newTab, newPrivateTab]]
+    }
+
+    private func getNewTabAction() -> PhotonRowActions {
+        return SingleActionViewModel(title: .KeyboardShortcuts.NewTab,
+                                     iconString: StandardImageIdentifiers.Large.plus,
+                                     iconType: .Image) { _ in
             let shouldFocusLocationField = self.newTabSettings == .blankPage
             self.overlayManager.openNewTab(url: nil, newTabSettings: self.newTabSettings)
             self.openBlankNewTab(focusLocationField: shouldFocusLocationField, isPrivate: false)
         }.items
+    }
 
-        let newPrivateTab = SingleActionViewModel(title: .KeyboardShortcuts.NewPrivateTab,
-                                                  iconString: StandardImageIdentifiers.Large.plus,
-                                                  iconType: .Image) { _ in
+    private func getNewPrivateTabAction() -> PhotonRowActions {
+        let isRefactorEnabled = isToolbarRefactorEnabled && isOneTapNewTabEnabled
+        return SingleActionViewModel(title: .KeyboardShortcuts.NewPrivateTab,
+                                     iconString: isRefactorEnabled ? StandardImageIdentifiers.Large.privateMode : StandardImageIdentifiers.Large.plus,
+                                     iconType: .Image) { _ in
             let shouldFocusLocationField = self.newTabSettings == .blankPage
             self.overlayManager.openNewTab(url: nil, newTabSettings: self.newTabSettings)
             self.openBlankNewTab(focusLocationField: shouldFocusLocationField, isPrivate: true)
-            TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .newPrivateTab, value: .tabTray)
+            // By checking whether the refactor is enabled, we will only record telemetry for old tab tray action
+            if !isRefactorEnabled {
```

Comment:
When the toolbar refactor is enabled we also need Telemetry. So for now I don't think we will need this check.

Reply:
Gotcha, was unsure if there were separate tickets for telemetry. Are there specs for telemetry here? Wondering what the `value` field should be when this event is triggered from the new tab long press action sheet

Reply:
We have [this task](https://mozilla-hub.atlassian.net/browse/FXIOS-8714) for the telemetry. It should have all the specs. But you don't have to address this now as we have the task for it.